src/components/button/Button.tsx

```diff
@@ -454,23 +57,30 @@ type ButtonBase = BaseProps & {
   variant?: Variant;
 };
 
+/** These types are a bit cobbled together to allow for the polymorphic tag names.
+ * It may be a good idea to look into refactoring this funcitonality
+ */
+
 export type GroupTag = 'div' | 'a' | 'button';
 
-/* These types are kind of hacked because styled components makes it difficult to do polymorphic components. */
+const DEFAULT_TAG = 'button' as const;
 
 type MyHtmlElement<T extends GroupTag> = T extends 'a'
   ? HTMLAnchorElement
-  : T extends 'div'
-  ? HTMLDivElement
-  : HTMLButtonElement;
-
-export type ButtonProps<T extends GroupTag = 'button'> = ButtonBase &
-  Omit<ComponentPropsWithoutRef<T>, keyof ButtonBase | 'onClick'> & {
+  : T extends 'button'
+  ? HTMLButtonElement
+  : HTMLElement;
+
+// source: https://stackoverflow.com/questions/55969769/typing-a-dynamic-tag-in-react-with-typescript#:~:text=I%20don%27t%20see,the%20div%20tag.
+export type ButtonProps<T extends GroupTag = typeof DEFAULT_TAG> =
+  ButtonBase & {
+    href?: T extends 'a' ? string : never;
     onClick?: MouseEventHandler<MyHtmlElement<T>>;
-    tag?: T;
-  };
+    tag?: T | GroupTag;
+  } & (Omit<ComponentPropsWithoutRef<T>, keyof ButtonBase | 'onClick'> &
+      HTMLAttributes<HTMLElement>);
```

Comment:
ComponentPropsWithoutRef<T> would already have all of the attributes for the `tag`. Would extending `HTMLAttributes<HTMLElement>` be necessary here?
```suggestion
  } & (Omit<ComponentPropsWithoutRef<T>, keyof ButtonBase | 'onClick'>;
```

Reply:
It appears to be necessary. When removing the `HTMLAttributes<HTMLElement>` type, the `Tag` "component" throws an error that the types of property `onCopy` are incompatible for the anchor and button elements. When solving for that, we get another similar error for a similar property. I went down a trail of solving for about 12 of these properties before I gave up. The `HTMLAttributes<HTMLElement>` resolves that.

But this is definitely a part I could be misunderstanding. I am open to being taught about it.